README.md

```diff
@@ -3,13 +3,29 @@
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
 
+  Asymptotics drops constants and lower order terms because asymptotics is focused on the big picture. As n gets larger and larger, lower order terms and exponents matter less, so when using asymptotics, we only need to care about the high order terms even though lower order terms have an impact on actual runtime with smaller values of n.
+  Asymptotics only gives general bounds for runtimes, as asymptotics provides answers as n gets infinitely large, meaning that could be a large difference in runtimes between two programs who both have the same asymptotic expression
+  Asymptotics considers only the theoretical runtimes of the algorithm in a ideal system without taking into account external factors. Memory issues, computational issues, factors intrinsic to the programming, or a machine which cannot handle the code would perform worse than the asymptotic expression would suggest.
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+  Since a binary search tree has a complexity of O(log(n)), so finding an element in a tree with 1,000 elements takes 5 seconds, or 1.6 * log(1000) = 5 seconds.
+   Which would mean that finding an element in a binary search tree with 10,0000 elements would take 1.6 * log(10,000) = 6.4 seconds.
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+  The constants used for calculation that are dropped for complexity may have a larger effect with a larger set of input data, which would be less important on a smaller tree, but could account for the time difference.
+  There could be various real-world issues like memory issues, CPU issues, or one of any number of bottlenecks or resource allocations in the machine that cause the program to run much slower.
+  The O(log(n)) complexity assumes a balanced tree. If the tree is unbalanced, it could be as bad as O(n), which would account for the different time.
```

Comment:
If it's linear, I would expect it to take 50 seconds, which is still much faster.